### PR TITLE
New ATC commands `profile list` and `profile dump`

### PR DESCRIPTION
**Pull request by mnezerka**
New ATC commands together with many lint fixes (separate commit).

### CODE REVIEW

**Comment by jfilak** [comment]
Well done! Thank you. I want to just run it on my machine before I merge it.

**Comment by jfilak** [comment]
Do you suppose the code will be used the following way:

``` bash
sapcli atc profiles
sapcli atc dump -p STANDARD | tee ATC_PROFILE_STANDARD.json
```

?

I am asking because I am just not quite sure I like the option name "dump"
What if we have it this way:

``` bash
sapcli atc profile list
sapcli atc profile dump -p STANDARD | tee ATC_PROFILE_STANDARD.json
```

**Comment by mnezerka** [comment]
Yes, you are right. My first idea was to implement it as a subcommands (`sapcli atc profile list,dump`), but I was not able to decode proper use of all those Command decorators to achieve this. If you give me hint how to implement subcommands, I'll be very happy to refactor it.

**Comment by mnezerka** [comment]
Thanks for support Jakub. I adapted my PR according to your instructions. Subcommand `profile` is implemented.

File: doc/commands/atc.md [hunk 1]
```diff
@@ -94,3 +94,113 @@ The default configuration in JSON is the following:
 ```json
 {"1": "error", "2": "error", "3": "warning", "4": "warning", "5": "info"}
 ```
+
+## profiles
```
Comment (jfilak) [inline doc/commands/atc.md:98]: You sure you want to keep it as `## profiles`?

File: doc/commands/atc.md [hunk 2]
```diff
@@ -94,3 +94,113 @@ The default configuration in JSON is the following:
...
+  "PROFILE2": {
+    "created": "20210309180000",
+    "created_by": "BOB",
+    "changed": "20080415161735",
+    "changed_by": "BOB"
+  }
+}
+```
+
+## dump
```
Comment (jfilak) [inline doc/commands/atc.md:143]: You sure you want to keep it as `## dump`?

File: doc/commands/atc.md (hunk 1 above)
Comment (jfilak) [reply doc/commands/atc.md:98]: Add it to the header please.

File: doc/commands/atc.md (hunk 2 above)
Comment (jfilak) [reply doc/commands/atc.md:143]: Add it to the header please.

**Review by jfilak — CHANGES_REQUESTED** [review]

**Comment by jfilak** [comment]
Thank you!

**Pull request merged**